### PR TITLE
Schemas: Allow custom blocks in theme.json styles

### PR DESCRIPTION
**Pull request by walbo**
## Description
This PR fixes the JSON Schema to add support for custom blocks in the styles blocks section of theme.json files. Support for custom blocks in the settings section was added in #36341

## How has this been tested?
```json
{
	"$schema": "../theme.json",
	"version": 1,
	"styles": {
		"blocks": {
			"core/archives": {
				"border": {
					"color": "#000"
				}
			},
			"example/namespace": {
				"border": {
					"color": "#000"
				}
			}
		}
	}
}
```

## Types of changes
Update JSON Schema to allow custom blocks

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Review by ryanwelcher — APPROVED** [review]
LGTM! 👍

File: schemas/json/theme.json [hunk 1]
```diff
@@ -781,258 +780,269 @@
...
-				},
-				"core/video": {
-					"$ref": "#/definitions/stylesPropertiesAndElementsComplete"
-				},
-				"core/widget-area": {
-					"$ref": "#/definitions/stylesPropertiesAndElementsComplete"
-				},
-				"core/legacy-widget": {
-					"$ref": "#/definitions/stylesPropertiesAndElementsComplete"
+			"allOf": [
```
Comment (ajlende) [inline schemas/json/theme.json]: Is `allOf` required here, or can `patternProperties` just be used alongside `properties`?

Comment (walbo) [reply schemas/json/theme.json]: Yes. That should work. I followed how it was done in `settings.blocks`.

I'll update the PR and change it in `settings.blocks` as well (for consistency).

Comment (walbo) [reply schemas/json/theme.json]: @ajlende Updated.

Testet locally that:
- autosuggest for core blocks still works
- custom blocks are allowed
- custom properties with invalid name shows warning

**Review by ajlende — APPROVED** [review]
Thanks for cleaning up that other spot too! Tested locally in VSCode and IntelliJ; LGTM 👍

**Review by aristath — APPROVED** [review]

**Comment by Mamaduka** [comment]
Backported to 5.9.1.

**Pull request merged**